templates/common/baremetal/files/NetworkManager-resolv-prepender.yaml

```diff
@@ -49,6 +47,37 @@ contents:
             show \
             "{{.Infra.Status.PlatformStatus.BareMetal.APIServerInternalIP}}" \
             "{{.Infra.Status.PlatformStatus.BareMetal.IngressIP}}")
+        # check if we got DHCPv6 hostname. The value would usually be in
+        # DHCP6_FQDN_FQDN but NM internal client does not parse the option
+        # Only if it is IPv4
+        if grep -q ":" "$NAMESERVER_IP"; then
+            LEASE_FILE=$(mktemp)
+            const CONT_NAME="dhcpv6-hostname-workaround"
+            /usr/bin/podman run --rm -d --name "$CONT_NAME" \
+                --authfile /var/lib/kubelet/config.json \
+                --net=host \
+                --entrypoint /usr/sbin/dhclient \
+                -v "${LEASE_FILE}:/tmp/workaround.lease:Z" \
+                {{ .Images.baremetalRuntimeCfgImage }} \
+                -d  -sf /bin/true -lf /tmp/workaround.lease -6 -v "$IFACE"
+            DHCPv6_DHCLIENT_TIMEOUT=15
+            while : ; do
+                mapfile -t NAME_DATA < <(awk '/(host|domain)-name / {print substr($3, 2, length($3) - 3)}' "$LEASE_FILE")
```

Comment:
Just for reference for anyone else looking at this, the part of the lease file that we care about looks something like this:

  option host-name "my-hostname";
  option domain-name "my.domain";

That helps make sense of some of the string processing that happens here.

Reply:
Yup!